src/main/java/org/cbioportal/security/CancerStudyPermissionEvaluator.java

```diff
@@ -284,6 +285,8 @@ private boolean hasAccessToCancerStudy(Authentication authentication, CancerStud
         // check if user is in study groups
         // performance now takes precedence over group accuracy (minimal risk to caching cancer study groups)
         Set<String> groups = new HashSet(Arrays.asList(cancerStudy.getGroups().split(";")));
```

Comment:
This is preference, but I would instead just filter out inline of creating the Set

`        Set<String> test = Arrays.stream(cancerStudy.getGroups().split(";"))
            .filter(String::isEmpty)
            .collect(Collectors.toSet());`
            
            
Same for the other one as well... Just incase someone adds something before this statement using groups and misses the removeIf statement